WooCommerce/Classes/Extensions/UINavigationController+Woo.swift

```diff
@@ -29,5 +29,44 @@ extension UINavigationController {
         popToRootViewController(animated: animated)
         CATransaction.commit()
     }
+}
+
+// MARK: - Handle UINavigationBar's 'Back' button action
+//
+protocol  UINavigationBarBackButtonHandler {
+
+    /// Should block the 'Back' button action
+    ///
+    /// - Returns: true - don't block，false - block
+    func  shouldPopOnBackButton() -> Bool
+}
+extension UIViewController: UINavigationBarBackButtonHandler {
+    //Do not block the "Back" button action by default, otherwise, override this function in the specified viewcontroller
+    @objc func  shouldPopOnBackButton() -> Bool {
+        return true
+    }
+}
+extension UINavigationController: UINavigationBarDelegate {
+    public func navigationBar(_ navigationBar: UINavigationBar, shouldPop item: UINavigationItem) -> Bool {
```

Comment:
😄 yes! It's the only not invasive way I found to take and manage the action from a system back button. The other way is to use a custom back button item, that I would avoid since it would create a precedent for not standardizing all the back buttons.